### PR TITLE
[libdevice] Add tests for imf simd APIs.

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @v-klochkov 
This PR is test for https://github.com/intel/llvm/pull/6327
Could you review it?
Thanks very much.

File: SYCL/DeviceLib/imf_simd_emulate_test.cpp [hunk 1]
```diff
@@ -0,0 +1,1835 @@
...
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <cassert>
+#include <iostream>
+
+namespace s = cl::sycl;
```
Comment (v-klochkov) [inline SYCL/DeviceLib/imf_simd_emulate_test.cpp]: There is ongoing activity to remove 'cl'. Please change it same way as it is done here: https://github.com/intel/llvm-test-suite/blob/intel/SYCL/DeviceLib/math_test.cpp#L15
```suggestion
namespace s = sycl;
```

File: SYCL/DeviceLib/imf_simd_emulate_test.cpp [hunk 2]
```diff
@@ -0,0 +1,1835 @@
...
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// RUN: %clangxx -fsycl -fno-builtin -fsycl-device-lib-jit-link %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
```
Comment (v-klochkov) [inline SYCL/DeviceLib/imf_simd_emulate_test.cpp]:
```suggestion
#include <sycl/sycl.hpp>
```

**Comment by jinge90** [comment]
Hi, @v-klochkov 
The legacy sycl header are replaced, could you review again?
Thanks very much.

**Review by v-klochkov — APPROVED** [review]

**Comment by pvchupin** [comment]
@jinge90, I assume it's not expected to work on cuda/hip?
Please disable there....

Currently fail at https://github.com/intel/llvm/runs/7847235161?check_suite_focus=true

**Comment by jinge90** [comment]
> @jinge90, I assume it's not expected to work on cuda/hip? Please disable there....
> 
> Currently fail at https://github.com/intel/llvm/runs/7847235161?check_suite_focus=true

Yes, we don't have plan to enable those APIs on CUDA/HIP.

**Pull request merged**